src/factories.js

```diff
@@ -50,21 +46,19 @@ export function createShorthand(Component, mapValueToProps, val, defaultProps =
 
   // Map values to props and create a ReactElement
   if (_.isString(val) || _.isNumber(val)) {
-    return <Component {...mergePropsAndClassName(defaultProps, mapValueToProps(val))} />
+    return <Component {...mergePropsAndClassName(mapValueToProps(val), defaultProps)} />
```

Comment:
@jcarbo I did in fact have to reverse the props for the string/number literal case only.  Otherwise, the default `type` could not be overridden for `createHTMLInput`.

```jsx
createHTMLInput(this.props.input, { type: 'password' })
```

In this case, if the user does not provide an `input` prop, then we want `type='password'`.  However, if the `mapValueToProps` argument always wins, then the `type: 'password'` will always be overridden with `type: undefined`.

This is true for only for the literal value usage.  The `mapValueToProps` is really like a `baseProps`, and is only invoked for the literal value case.  It should also have less priority than the `defaultProps`, which then have less priority than the user's prop object/element props.